pages/map/index.tsx

```diff
@@ -1,109 +1,69 @@
+import Message from "../../components/map/Message";
+import PlaceCard from "../../components/map/PlaceCard";
 
-interface Coords {
-    lat: number,
-    lng: number
-};
-
-interface MapProps {
-    center: Coords,
-    children?: React.ReactNode
-};   
-
-interface MarkerProps {
-    color: string
-}
-
-const LandingMessage = () => {
-    const [alert, setAlert] = useState(true);
-
-    return(
-        <Alert variant="info" show={alert} dismissible onClose={() => setAlert(false)}>
-            <Alert.Heading>Welcome To The Map</Alert.Heading>
-            <p>
-                Here you can see location and other information of recycling centers!
-                You can also see your saved recycling centers for convenience!
-            </p>
-            <hr/>
-            <p>Enter a zip code or use your current location to get started searching...</p>
-        </Alert>
-    );
-}
-
-const Marker = ({color}: MarkerProps) => (
-    <FaMapMarkerAlt
-        color={color}
-        size={36}
-        className={styles.marker}
-    />
-);
-
-const Map = ({center, children}: MapProps) => {
-    const defaultCoords: Coords = {lat: 40.7128, lng: -74.0060};
-    const mapOptions = {
-        mapTypeControl: true,
-        streetViewControl: true
-    };
-
-    let isNull = Object.values(center).every(obj => obj === null);
-    
-    return(
-        <div className={`${styles.map} shadow`}>
-            <GoogleMap
-                //@ts-ignore
-                bootstrapURLKeys={{key: mapsKey, libraries: ["geometry", "drawing"]}}
-                defaultCenter={defaultCoords}
-                center={isNull ? defaultCoords : center}
-                defaultZoom={11}
-                options={mapOptions}
-                yesIWantToUseGoogleMapApiInternals
-            >
-                {children}   
-            </GoogleMap>
-        </div>        
-    );
-}
+import { getPlaceData, getPlaceDataWithZip } from "../../components/api";
 
 const MapPageContent = () => {
     const [zip, setZip] = useState("");
+    const [isZip, setIsZip] = useState(false);
+
     const [userCoords, setUserCoords] = useState({lat: null, lng: null});
+    const [focusedMarker, setFocusedMarker] = useState({lat: null, lng: null});
+    
+    const [isFocused, setIsFocused] = useState(false);
     const [places, setPlaces] = useState([]);
-    const [zipStatus, setZipStatus] = useState(false)
-
+    
     const changeHandler = (e: any): void => {
         e.preventDefault();
         setZip(e.target.value);
     };
 
+    const focusHandler = (e: any, coords: Coords) => {
+        e.preventDefault();
+        setFocusedMarker((prev: any) => {
+            return {
+                ...prev,
+                lat: coords.lat,
+                lng: coords.lng
+            }
+        });
+
+        setIsFocused(true);
+
+        document.body.scrollTop = 0;
+        document.documentElement.scrollTop = 0;
+    }
+
     const submitHandler = (e: any) => {
```

Comment:
unlike the other two handlers, `submitHandler` is a reasonable name since it's easily inferred that the component has a form. forms are generally the only thing with "submit" behavior, whereas many elements support "change" and "focus" behavior.